communication/src/main/java/org/jboss/da/communcation/pnc/authentication/PNCAuthFilter.java

```diff
@@ -0,0 +1,25 @@
+package org.jboss.da.communcation.pnc.authentication;
+
+import org.jboss.da.communcation.pnc.authentication.PNCAuthentication;
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import java.io.IOException;
+
+/**
+ * Filter to add authorization header to all http requests to PNC
+ */
+public class PNCAuthFilter implements ClientRequestFilter {
+
+    /**
+     * It works by add the Authorization key, followed by "Bearer <token>"
+     * to the http request
+     *
+     * @param requestContext
+     * @throws IOException
+     */
+    public void filter(ClientRequestContext requestContext) throws IOException {
+        String token = PNCAuthentication.authenticate();
```

Comment:
How long the token is valid? Do we need to authenticate again for every request or can we reuse the token?